source/credentials_provider_sts.c

```diff
@@ -229,6 +234,40 @@ static int s_sts_xml_on_AssumeRoleResult_child(struct aws_xml_node *node, void *
     if (aws_byte_cursor_eq_c_str_ignore_case(&node_name, "Credentials")) {
         return aws_xml_node_traverse(node, s_sts_xml_on_Credentials_child, user_data);
     }
+    if (aws_byte_cursor_eq_c_str_ignore_case(&node_name, "AssumedRoleUser")) {
+        return aws_xml_node_traverse(node, s_sts_xml_on_AssumedRoleUser_child, user_data);
+    }
+
+    return AWS_OP_SUCCESS;
+}
+
+static int s_sts_xml_on_AssumedRoleUser_child(struct aws_xml_node *node, void *user_data) {
+    struct sts_creds_provider_user_data *provider_user_data = user_data;
+    struct aws_byte_cursor node_name = aws_xml_node_get_name(node);
+    struct aws_byte_cursor arn_cursor;
+    AWS_ZERO_STRUCT(arn_cursor);
+
+    if (aws_byte_cursor_eq_c_str_ignore_case(&node_name, "Arn")) {
+        if (aws_xml_node_as_body(node, &arn_cursor)) {
+            return AWS_OP_ERR;
+        }
+        struct aws_byte_cursor account_id;
+        AWS_ZERO_STRUCT(account_id);
+        /* The format of the Arn is arn:partition:service:region:account-id:resource-ID and we need to parse the
+         * account-id out of it which is the fifth element. */
+        for (int i = 0; i < 5; i++) {
+            if (!aws_byte_cursor_next_split(&arn_cursor, ':', &account_id)) {
```

Comment:
note: its valid for arn not to have account id field. most fields are optional from what i remember (s3 is an example of arns without a bunch of fields). do we want to have logic that accounts for that? or just setting it to empty string is fine?

Reply:
account_id is an optional nice-to-have field, so it's fine if it's blank.